src/python/tests/test_statistics_history.py

```diff
@@ -0,0 +1,150 @@
+        select_stats_history_value_stmt = sa.select([_STATS_HISTORY_TABLE.c.value]).where(
+            _STATS_HISTORY_TABLE.c.key == key)
+
+        async with aiopg.sa.create_engine(_CONNECTION_STRING) as engine:
+            async with engine.acquire() as conn:
+                async for result in conn.execute(select_stats_history_value_stmt):
+                    assert result[0] == rand_value
+
+    async def test_update_previous_value(self):
+        """
+        Test the update of statistics.previous_value
+        :assert:
+            Assert  statistics.previous_value  gets updated with the set value
+        """
+        key = random.choice(_KEYS)
+        rand_value = random.randint(1, 10)
+        _update_previous_value(key=key, value=rand_value)
+
+        select_previous_value_stmt = sa.select([_STATS_TABLE.c.previous_value]).where(
+            _STATS_TABLE.c.key == key)
+
+        async with aiopg.sa.create_engine(_CONNECTION_STRING) as engine:
+            async with engine.acquire() as conn:
+                async for result in conn.execute(select_previous_value_stmt):
+                    assert result[0] == rand_value
+
+    async def test_select_from_statistics(self):
+        """
+        Test that _select_from_statistics retrieves value and previous value from statistics table
+        :assert:
+            Assert that both value and previous value == 0
+        """
+        for key in _KEYS:
+            result = _select_from_statistics(key=key)
+            assert result[0][0] == 0
+            assert result[0][0] == result[0][1]
+
+    async def test_stats_history_main(self):
+        """
+        Test that as a whole (statistics_history_main) executes properly
+        :assert:
+            Assert that once stats_history_main() is done, both statistics,
+            and statistics_history are up to date.
+        """
+        random_key = random.choice(_KEYS)
+        await update_statistics_value(statistics_key=random_key,
+                                      value_increment=random.randint(1, 10))
+        stats_history_main()
+
+        select_stats_stmt = sa.select([_STATS_TABLE.c.value, _STATS_TABLE.c.previous_value]).where(
+            _STATS_TABLE.c.key == random_key)
+        select_stats_history_stmt = sa.select([_STATS_HISTORY_TABLE.c.value]).where(
+            _STATS_HISTORY_TABLE.c.key == random_key)
+        async with aiopg.sa.create_engine(_CONNECTION_STRING) as engine:
+            async with engine.acquire() as conn:
+                # Execute select_stats_stmt
+                async for result in conn.execute(select_stats_stmt):
+                    select_stats = [result[0], result[1]]
+                # Execute select_stats_history_stmt
+                async for result in conn.execute(select_stats_history_stmt):
+                    select_stats_history = result[0]
+
+        assert select_stats[0] == select_stats[1] == select_stats_history
```

Comment:
pycharm error: Local variable 'select_stats' and 'select_stats_history' might be referenced before assignment.